modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipServiceImpl.java

```diff
@@ -196,6 +190,33 @@ public ObjectRelationship updateObjectRelationship(
 			deletionType, edge, labelMap, objectField);
 	}
 
+	private void _checkUpdatePermission(
+			long objectDefinitionId, long primaryKey)
+		throws PortalException {
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionPersistence.findByPrimaryKey(objectDefinitionId);
+
+		if ((objectDefinition != null) && objectDefinition.isSystem()) {
```

Comment:
you don't need to check if it is different than null because of the method find before. check if it is `objectDefinition.isUnmodifiableSystemObject()`